share/blob_tx.go

```diff
@@ -0,0 +1,78 @@
+package share
+
+import (
+	"errors"
+
+	v1 "github.com/celestiaorg/go-square/proto/blob/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	// ProtoBlobTxTypeID is included in each encoded BlobTx to help prevent
+	// decoding binaries that are not actually BlobTxs.
+	ProtoBlobTxTypeID = "BLOB"
+)
+
+type BlobTx struct {
+	Tx    []byte
+	Blobs []*Blob
+}
+
+// UnmarshalBlobTx attempts to unmarshal a transaction into blob transaction. If an
+// error is thrown, false is returned.
+func UnmarshalBlobTx(tx []byte) (*BlobTx, bool, error) {
```

Comment:
[nit] I think providing a name for the second return parameter would help communicate it's meaning

```suggestion
func UnmarshalBlobTx(tx []byte) (blobTx *BlobTx, isBlobTx bool, err error) {
```

Reply:
I'm just going to try make it clear in the comments